build/ci-image/Dockerfile

```diff
@@ -4,10 +4,16 @@ RUN dnf -y update && \
   dnf -y install git golang python3 python3-pip python3-pip-wheel make && \
   dnf clean all
 RUN pip3 install pre-commit
-RUN go get -u golang.org/x/lint/golint
 
+RUN useradd -u 1001 -ms /bin/bash test-user && mkdir /home/test-user/receptor \
+    && chown test-user /home/test-user/receptor
+
+USER test-user
+
+RUN go get -u golang.org/x/lint/golint
 # Warm up the Go build cache
-RUN git clone https://github.com/project-receptor/receptor && cd receptor && make build-all && cd .. && rm -rf receptor
+RUN cd /home/test-user && git clone https://github.com/project-receptor/receptor receptor-tmp && cd receptor-tmp && make build-all && cd .. && rm -rf receptor-tmp
 
-COPY entrypoint.sh /entrypoint.sh
-ENTRYPOINT ["/entrypoint.sh"]
+WORKDIR /home/test-user/receptor
+COPY entrypoint.sh /home/test-user/entrypoint.sh
+ENTRYPOINT ["/home/test-user/entrypoint.sh"]
```

Comment:
Why move the entry point?

Reply:
just for consistency sake of having everything in the test-user's homedir, we can move it back if that's preferable

Reply:
It seems strange to me to think of the container entrypoint as associated with a user.  I'd rather leave it where it is.

Reply:
yes that is the reason for it, and i dont know if they explicitly say they make that guarantee (I couldnt find anything about the user in the docs and did some print debugging to figure out the uid) but this is the only way i could figure out to make this work. It seems unlikely that they'll update the user but the totally could